Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,75 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
                 }
             }
 
-            return new PlanarSurface { ExternalBoundary = externalBoundary, InternalBoundaries = internalBoundaries };
+            //// Internal checks independent from External
+            // Selfintersections Internal
+            count = internalBoundaries.Count;
+
+            internalBoundaries = internalBoundaries.Where(x => x is NurbsCurve || !x.IIsSelfIntersecting()).ToList();
+
+            if (internalBoundaries.Count != count)
+            {
+                Reflection.Compute.RecordWarning("At least one of the internalBoundaries is selfintersecting. And have been disregarded.");
+            }
+
+            // Runs Boolean Union to resolve intersections and containemnt
+            count = internalBoundaries.Count;
+
+            internalBoundaries = internalBoundaries.Where(x => x is NurbsCurve || x is Ellipse)
+                                .Concat(internalBoundaries.Where(x => !(x is NurbsCurve) && !(x is Ellipse)).BooleanUnion()).ToList();
+
+            if (internalBoundaries.Count != count)
+            {
+                Reflection.Compute.RecordWarning("At least one of the internalBoundaries was overlapping another one. BooleanUnion was used to resolve it.");
+            }
+
+            //// Checks based around external curve
```

Comment:
Wouldn't it be better to do the external curve checks at the beginning?

Reply:
I wanted to save what could be saved from the internal curves before returning the planarSurface, which happens here if it's a Nurbs/Ellipse.

Could do another check in the beginning for selfintersections ignoring Nurbs and such, but struck me as even more messy

Reply:
Makes sense, I commented over myself in my latest review.